auth/index.js

```diff
@@ -17,6 +17,32 @@ exports.isValidPassword = function (password, hashedPassword) {
 exports.doLogin = handler.doLogin
 
 exports.middleware = async function authMiddleware(req, res, next) {
+  const key = req.query['access_key'] || req.headers['x-access-key']
+  if (key) {
+    if (typeof key == 'string') {
+      const accessKeys = await req.db['access_keys'].find({ key: { $eq: key } })
+      if (accessKeys && accessKeys.length > 0) {
+        const accessKey = accessKeys[0]
+        // check if access key is still valid
+        if (accessKey.expires_at >= new Date().toISOString()) {
+          let user
+          try {
+            user = await req.db['users'].get(accessKey.user_id)
```

Comment:
once `user_collection` added to schema this will become `req.db[accessKey.user_collection]`